### PR TITLE
docs(misc): Updates and Fixes to Package-Based tutorial

### LINKED ISSUE #12689: Package-Based Repo Tutorial Does Not Work:  NX   Cannot find project 'is-even'
## Current Behavior
The tutorial found here: https://nx.dev/getting-started/package-based-repo-tutorial does not work.

After working through the tutorial adding the `is-even` package as expected, the `npx nx build is-even` command fails with the following message:
```
> NX   Cannot find project 'is-even'
```

## Expected Behavior

The tutorial found here: https://nx.dev/getting-started/package-based-repo-tutorial should work.

## Steps to Reproduce

Simply follow the steps in the tutorial here: https://nx.dev/getting-started/package-based-repo-tutorial 

### Failure Logs

```
> NX   Cannot find project 'is-even'
```

### Environment

```
> NX   Cannot find project 'is-even'
```

### PR DESCRIPTION
**Pull request by ZackDeRose**
## Current Behavior

+ Package-based tutorial doesn't work correctly using snippets provided.
+ Some instructions unclear, and cache doesn't behave as expected given the chain of events.

## Expected Behavior

+ Tutorial behaves as described.
+ Correct caching results reflected.

## Related Issue(s)

Related to: #12859
Fixes: #12689

### CODE REVIEW

File: docs/shared/npm-tutorial/package-based.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ npx create-nx-workspace@latest package-based --preset=npm
 The file structure should look like this:
 
 ```treeview
-myorg/
+package-based/
```
Comment (ZackDeRose) [inline docs/shared/npm-tutorial/package-based.md:14]: fixes from older project name

File: docs/shared/npm-tutorial/package-based.md [hunk 2]
```diff
@@ -22,19 +22,16 @@ myorg/
...
-
-```shell
-npm i typescript -D -W
+package-based/
+├── packages/
+│   └── is-even/
+│       ├── index.ts
+│       └── package.json
+├── nx.json
+└── package.json
```
Comment (ZackDeRose) [inline docs/shared/npm-tutorial/package-based.md:31]: "full" treeview for better context on the nesting of the `is-even` package code

File: docs/shared/npm-tutorial/package-based.md [hunk 3]
```diff
@@ -61,27 +58,37 @@ export const isEven = (x: number) => x % 2 === 0;
 {% /tab %}
 {% /tabs %}
 
-Update the `.gitignore` file to make sure you have a `dist` entry (without a leading slash) to ignore the `dist` folder inside the `is-even` package.
+Next install TypeScript (notice we're using `tsc` for the `build` script in `package.json` above). While we could install TypeScript at the package-level, it is more convenient to have it globally for the entire monorepo. Run the following command at the root of your workspace.
+
+```shell
+npm i typescript -D -W
+```
```
Comment (ZackDeRose) [inline docs/shared/npm-tutorial/package-based.md:65]: Moved installing typescript until after the `packages/is-even/package.json` file is populated correctly.

(With workspaces turned on, was getting an error here when trying to run `npm i` with this file empty)

File: docs/shared/npm-tutorial/package-based.md [hunk 4]
```diff
@@ -61,27 +58,37 @@ export const isEven = (x: number) => x % 2 === 0;
...
+package-based/
+├── packages/
+│   ├── is-even/
+│   │   ├── index.ts
+│   │   └── package.json
+│   └── is-odd/
+│       ├── index.ts
+│       └── package.json
+├── nx.json
+└── package.json
```
Comment (ZackDeRose) [inline docs/shared/npm-tutorial/package-based.md:91]: "Full" treeview to better understand at a glance how the nesting works.

**Review by bcabanes — APPROVED** [review]

File: docs/shared/npm-tutorial/package-based.md [hunk 5]
```diff
@@ -48,7 +45,7 @@ export const isEven = (x: number) => x % 2 === 0;
 
 ```json {% fileName="packages/is-even/package.json" %}
 {
-  "name": "@package-based/is-even",
+  "name": "is-even",
```
Comment (ZackDeRose) [inline docs/shared/npm-tutorial/package-based.md:48]: removed scope

File: docs/shared/npm-tutorial/package-based.md [hunk 6]
```diff
@@ -100,33 +107,33 @@ export const isOdd = (x: number) => !isEven(x);
 
 ```json {% fileName="packages/is-odd/package.json" %}
 {
-  "name": "@package-based/is-odd",
+  "name": "is-odd",
```
Comment (ZackDeRose) [inline docs/shared/npm-tutorial/package-based.md:110]: removed scope

File: docs/shared/npm-tutorial/package-based.md [hunk 7]
```diff
@@ -100,33 +107,33 @@ export const isOdd = (x: number) => !isEven(x);
...
   "version": "0.0.0",
   "main": "dist/index.js",
   "devDependencies": {},
   "scripts": {
     "build": "tsc index.ts --outDir dist"
   },
   "dependencies": {
-    "@package-based/is-even": "*"
+    "is-even": "*"
```
Comment (ZackDeRose) [inline docs/shared/npm-tutorial/package-based.md:118]: removed scope to match `is-even`'s `name` value in `package.json`

File: docs/shared/npm-tutorial/package-based.md [hunk 8]
```diff
@@ -100,33 +107,33 @@ export const isOdd = (x: number) => !isEven(x);
...
 
-At the root of your workspace run
+At the root of your workspace run:
 
 ```shell
 npm install
 ```
 
-This allows the NPM workspace to properly link the new `is-odd` package.
+NPM will create a Symbolic Link in your file system at: `node_modules/is-even` and `node_modules/is-odd`, so they reflect changes to your `packages/is-even` and `packages/is-odd` directories as they happen.
```
Comment (ZackDeRose) [inline docs/shared/npm-tutorial/package-based.md:136]: Clarification of the symlinking that npm workspaces uses.

File: docs/shared/npm-tutorial/package-based.md [hunk 9]
```diff
@@ -153,59 +160,77 @@ Remove any existing `dist` folder and run:
...
-To build the `is-even` package run:
-
-```shell
-npx nx build is-even
-```
-
-Run the same command a second time and you'll see the build cache is being used:
+Run the command:
 
 ```{% command="npx nx build is-even" %}
```
Comment (ZackDeRose) [inline docs/shared/npm-tutorial/package-based.md:169]: Cache is already warmed - skipping first run.

File: docs/shared/npm-tutorial/package-based.md [hunk 10]
```diff
@@ -153,59 +160,77 @@ Remove any existing `dist` folder and run:
...
-```shell
-npx nx run-many --target=build
-```
-
-What you would get is the following:
-
 ```{% command="npx nx run-many --target=build" %}
     ✔  nx run is-even:build  [existing outputs match the cache, left as is]
-    ✔  nx run is-odd:build (906ms)
+    ✔  nx run is-odd:build  [existing outputs match the cache, left as is]
```
Comment (ZackDeRose) [inline docs/shared/npm-tutorial/package-based.md:193]: cache for `is-odd` is already warmed as we had the users build `is-odd` in the prior step.

File: docs/shared/npm-tutorial/package-based.md [hunk 11]
```diff
@@ -153,59 +160,77 @@ Remove any existing `dist` folder and run:
...
+    ✔  nx run is-odd:build (1s)
 
-```shell
-npx nx affected --target=build
+ ———————————————————————————————————————————————————————————————————————
+
+ >  NX   Successfully ran target build for 2 projects (2s)
 ```
 
+Notice that using this method, the `is-even` build ran before the `is-odd` build, and that the `is-even` build only happened once. This demonstrates how `run-many` is informed by the `targetDefaults` we set earlier.
```
Comment (ZackDeRose) [inline docs/shared/npm-tutorial/package-based.md:213]: Added demonstration of `run-many` leveraging the task graph.

**Pull request merged**